src/super_gradients/training/utils/callbacks/callbacks.py

```diff
@@ -863,3 +864,89 @@ def _infer_global_step(self, context: PhaseContext, is_train_loader: bool):
+
+
+def create_lr_scheduler_callback(
+    lr_mode: Union[str, Mapping],
+    train_loader: DataLoader,
+    net: torch.nn.Module,
+    training_params: Mapping,
+    update_param_groups: bool,
+    optimizer: torch.optim.Optimizer,
+) -> PhaseCallback:
+    """
+    Creates the phase callback in charge of LR scheduling, to be used by Trainer.
+
+    :param lr_mode: Union[str, Mapping],
+
+                    When str:
+
+                    Learning rate scheduling policy, one of ['step','poly','cosine','function'].
+
+                    'step' refers to constant updates at epoch numbers passed through `lr_updates`. Each update decays the learning rate by `lr_decay_factor`.
+
+                    'cosine' refers to the Cosine Anealing policy as mentioned in https://arxiv.org/abs/1608.03983.
+                      The final learning rate ratio is controlled by `cosine_final_lr_ratio` training parameter.
+
+                    'poly' refers to the polynomial decrease: in each epoch iteration `self.lr = self.initial_lr * pow((1.0 - (current_iter / max_iter)), 0.9)`
+
+                    'function' refers to a user-defined learning rate scheduling function, that is passed through `lr_schedule_function`.
+
+
+
+                    When Mapping, refers to a torch.optim.lr_scheduler._LRScheduler, following the below API:
+
+                        lr_mode = {LR_SCHEDULER_CLASS_NAME: {**LR_SCHEDULER_KWARGS, "phase": XXX, "metric_name": XXX)
+
+                        Where "phase" (of Phase type) controls when to call torch.optim.lr_scheduler._LRScheduler.step().
+
+                        The "metric_name" refers to the metric to watch (See docs for "metric_to_watch" in train(...)
+                         https://docs.deci.ai/super-gradients/docstring/training/sg_trainer.html) when using
+                          ReduceLROnPlateau. In any other case this kwarg is ignored.
+
+                        **LR_SCHEDULER_KWARGS are simply passed to the torch scheduler's __init__.
+
+
+
+
+    :param train_loader: DataLoader, the Trainer.train_loader used for training.
+
+    :param net: torch.nn.Module, the Trainer.net used for training.
+
+    :param training_params: Mapping, Trainer.training_params.
+
+    :param update_param_groups:bool,  Whether the Trainer.net has a specific way of updaitng its parameter group.
+
+    :param optimizer: The optimizer used for training. Will be passed to the LR callback's __init__
+     (or the torch scheduler's init, depending on the lr_mode value as described above).
+
+    :return: a PhaseCallback instance to be used by Trainer for LR scheduling.
+    """
+
+    if isinstance(lr_mode, str) and lr_mode in LR_SCHEDULERS_CLS_DICT:
+        sg_lr_callback_cls = LR_SCHEDULERS_CLS_DICT[lr_mode]
+        sg_lr_callback = sg_lr_callback_cls(
+            train_loader_len=len(train_loader),
+            net=net,
+            training_params=training_params,
+            update_param_groups=update_param_groups,
```

Comment:
Since `update_param_groups` can be easily inferred from the `net` itself via `update_param_groups = hasattr(unwrap_model(self.net), "update_param_groups")` we can safely drop this parameter and infer it's value inside the callback implementation itself.
I think this can simplify the code here.

Reply:
True, but rather keep this out of this PR.
update_param_groups is only used by our SG scheduler callbacks, so I don't feel like this change (although needed) should be here as it requires changing (and perhaps breaking code) of our LR scheduler callbacks.